### PR TITLE
docs: add comments and default value for configuration

### PR DESCRIPTION
**Pull request by yuhao900914**
### Summary

Add comments and the default value for configuration

Screenshot 2024-05-15 at 3 49 49 PM

### Checklist

* [ ] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:

### CODE REVIEW

**Review by izaaz — APPROVED** [review]

File: packages/analytics-types/src/config/browser.ts [hunk 1]
```diff
@@ -7,17 +7,64 @@ import { PageTrackingOptions } from '../page-view-tracking';
...
+   */
   partnerId?: string;
+  /**
+   * The custom Session ID for the current session.
+   * @defaultValue `timestamp`
+   */
   sessionId?: number;
+  /**
+   * The period of inactivity from the last tracked event before a session expires in milliseconds.
+   * @defaultValue `1800000`
```
Comment (Mercy811) [inline packages/analytics-types/src/config/browser.ts:47]: can we add delimiter for readability? 
```suggestion
   * @defaultValue `1,800,000` (30 minutes)
```

File: packages/analytics-types/src/config/browser.ts [hunk 2]
```diff
@@ -26,34 +73,97 @@ interface InternalBrowserConfig {
   lastEventTime?: number;
   lastEventId?: number;
   transportProvider: Transport;
-  pageCounter?: number;
```
Comment (Mercy811) [inline packages/analytics-types/src/config/browser.ts:29]: Is there any cases where customers need to set it at `init()`? If no, we should keep it as internal.

**Review by Mercy811 — APPROVED** [review]
Thanks Alyssa for adding in-editor comments. This is really useful.

We can also add dev doc links to each configuration for users need more details like [this](https://amplitude.slack.com/archives/C04HSDQA2VB/p1712607841384209?thread_ts=1712521462.343889&cid=C04HSDQA2VB).

File: packages/analytics-types/src/config/browser.ts (hunk 2 above)
Comment (yuhao900914) [reply packages/analytics-types/src/config/browser.ts:29]: Nope. However, this is bundled with page view plugin. We set the page counter in the page view plugin, we cannot access pageCounter if it's internal.

File: packages/analytics-types/src/config/browser.ts (hunk 1 above)
Comment (yuhao900914) [reply packages/analytics-types/src/config/browser.ts:47]: Updated.

**Comment by yuhao900914** [comment]
> Thanks Alyssa for adding in-editor comments. This is really useful.
> 
> We can also add dev doc links to each configuration for users need more details like [this](https://amplitude.slack.com/archives/C04HSDQA2VB/p1712607841384209?thread_ts=1712521462.343889&cid=C04HSDQA2VB).


Added in this commit.

**Pull request merged**